src/models/user_integration.rs

```diff
@@ -0,0 +1,111 @@
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use uuid::Uuid;
+
+use crate::{
+    database, schema::user_integrations as user_integration_schema, services::user as user_service,
+};
+
+#[derive(Queryable, Clone)]
+#[diesel(table_name = user_integration_schema)]
+struct UserIntegration {
+    id: Uuid,
+    related_user: Uuid,
+    name: String,
+    time: NaiveDateTime,
+}
+
+#[derive(Insertable, Clone)]
+#[diesel(table_name = user_integration_schema)]
+struct NewUserIntegration {
+    related_user: Uuid,
+    name: String,
+    time: NaiveDateTime,
+}
+
+impl user_service::NewUserIntegration {
+    fn to_model(&self) -> NewUserIntegration {
+        NewUserIntegration {
+            related_user: self.related_user.clone(),
+            name: self.name.clone(),
+            time: self.time,
+        }
+    }
+}
+
+impl UserIntegration {
+    fn to_service(&self) -> user_service::UserIntegration {
+        user_service::UserIntegration {
+            id: self.id.clone(),
+            related_user: self.related_user.clone(),
+            name: self.name.clone(),
+            time: self.time,
+        }
+    }
+}
```

Comment:
Don't you think that the service should be aware of the model but not vice versa? You can implement this in the service layer.

Reply:
Fixed.